### PR TITLE
feat: Added Personalize/Recommendations to Gen App Builder Demo

### PR DESCRIPTION
**Pull request by holtskinner**
- Added new page for https://cloud.google.com/generative-ai-app-builder/docs/try-personalize

### CODE REVIEW

File: gen-app-builder/search-web-app/genappbuilder_utils.py [hunk 1]
```diff
@@ -13,13 +13,38 @@
 # limitations under the License.
 
 """Generative AI App Builder Utilities"""
-from typing import List, Tuple
+from os.path import basename
+from typing import List, Optional, Tuple
 
-from google.cloud import discoveryengine
+from google.cloud import discoveryengine, discoveryengine_v1beta
```
Comment (koverholt) [inline gen-app-builder/search-web-app/genappbuilder_utils.py:19]: Is it dangerous to use objects/instances from `discoveryengine` and `discoveryengine_v1beta` at the same time? Might a user run into conflicts or compatibility issues? Maybe it's better to import only the `discoveryengine_v1beta` namespace and use everything from there? Or if not, then you can safely ignore this comment. 😁

File: gen-app-builder/search-web-app/main.py [hunk 2]
```diff
@@ -23,10 +23,15 @@
     PROJECT_ID,
     VALID_LANGUAGES,
     WIDGET_CONFIGS,
+    PERSONALIZE_DATASTORE_IDs,
 )
 from ekg_utils import search_public_kg
 from flask import Flask, render_template, request
-from genappbuilder_utils import search_enterprise_search
+from genappbuilder_utils import (
```
Comment (koverholt) [inline gen-app-builder/search-web-app/main.py:30]: Do you have a step listed somewhere in the README or elsewhere to install this Python package from somewhere? Also, is this Python package available on PyPI to be `pip install`ed?

File: gen-app-builder/search-web-app/main.py [hunk 3]
```diff
@@ -41,12 +46,23 @@
     {"link": "/", "name": "Gen App Builder - Widgets", "icon": "widgets"},
     {
         "link": "/search",
-        "name": "Gen App Builder - Custom UI",
+        "name": "Enterprise Search - Custom UI",
```
Comment (koverholt) [inline gen-app-builder/search-web-app/main.py:49]: 😎

File: gen-app-builder/search-web-app/static/recommend.js
Comment (koverholt) [inline gen-app-builder/search-web-app/static/recommend.js:1]: Is this an original file, or from another OSS library/repo? Might be useful to point out the source, and I'm curious if we can relicense this under Apache 2.0 (but I'm also not sure of the original source).

File: gen-app-builder/search-web-app/templates/recommend.html [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+      <p>
+        Made by <a href="https://github.com/holtskinner">holtskinner@</a> -
+        Source on
+        <a
+          href="https://github.com/GoogleCloudPlatform/generative-ai/tree/main/gen-app-builder/search-web-app"
+          >GitHub</a
+        >
+      </p>
+    </section>
+  </body>
```
Comment (koverholt) [inline gen-app-builder/search-web-app/templates/recommend.html:285]: (Optional) For folks that are only using the app and not looking at this repo/README, maybe add a callout for `Powered by Google Cloud` or similar with a link to Gen App Builder or the `Google Cloud Products Used` that are listed in the README.

**Review by koverholt — APPROVED** [review]
Nice, thanks for the update. I am a heavy user of this demo app, and lots of developers love seeing it in action!

File: gen-app-builder/search-web-app/genappbuilder_utils.py (hunk 1 above)
Comment (holtskinner) [reply gen-app-builder/search-web-app/genappbuilder_utils.py:19]: Valid point, I wasn't too sure about it. Wanted to encourage people to use the GA (v1) components whenever possible, but some features only support v1beta

File: gen-app-builder/search-web-app/templates/recommend.html (hunk 4 above)
Comment (holtskinner) [reply gen-app-builder/search-web-app/templates/recommend.html:285]: I've considered adding that, also been getting lots of questions about the datasets used, etc. I'll go ahead and add it.

File: gen-app-builder/search-web-app/static/recommend.js
Comment (holtskinner) [reply gen-app-builder/search-web-app/static/recommend.js:1]: It's my own file

File: gen-app-builder/search-web-app/main.py (hunk 2 above)
Comment (holtskinner) [reply gen-app-builder/search-web-app/main.py:30]: That's one of the local files

File: gen-app-builder/search-web-app/main.py
[the remark this answers is not in the document]
Comment (holtskinner) [reply gen-app-builder/search-web-app/main.py:1]: Figured it out, it was a key error for `cse_thumbnail`. Added a "Image not found" default image.

**Pull request merged**